### ISSUE TITLE
[css-content] Fix syntaxes according to CSS Values and Units grammar

### ISSUE THREAD

**Issue by lahmatiy** [w3c/csswg-drafts#663]
Comments for https://drafts.csswg.org/css-content/

## \<content-list\>

- `leader()` should be surrounded by angle brackets instead of quotes

```
  [ <string> | contents | <image> | <quote> | <target> | 'leader()' ]+
->
  [ <string> | contents | <image> | <quote> | <target> | <leader()> ]+
```

## leader( \<leader-type\> )

- Incorrect definition:
  - functions are declaring as `<function()>`
  - arguments are usually omitting -> `<leader-type>` should be omitted or declared aside
- A space before closing parenthesis is missing
- No semicolon at the end

```
  <leader-type> = leader( dotted | solid | space | <string>);
->
  <leader()> = leader( dotted | solid | space | <string> )
or
  <leader()> = leader( <leader-type> )
  <leader-type> = dotted | solid | space | <string> 
```

## \<target\>

- All functions should be surrounded by angle brackets
- Square brackets aren't needed

```
  <target> = [ target-counter() | target-counters() | target-text() ]
->
  <target> = <target-counter()> | <target-counters()> | <target-text()>
```

## \<target-counter()\>

- Angle brackets are missing in syntax reference
- Last group can be omitted since [comma are implicitly omissible](https://www.w3.org/TR/css3-values/#component-types)

```
  target-counter() = target-counter( [ <string> | <url> ] , <custom-ident> [ , <counter-style> ]? )
->
  <target-counter()> = target-counter( [ <string> | <url> ] , <custom-ident> , <counter-style>? )
```

## \<target-counters()\>

- Angle brackets are missing in syntax reference
- Missed `s` in function's name
- Last group can be omitted since [comma are implicitly omissible](https://www.w3.org/TR/css3-values/#component-types)

```
  target-counters() = target-counter(  [ <string> | <url> ] , <custom-ident> , <string> [ , <counter-style> ]? )
->
  <target-counters()> = target-counters(  [ <string> | <url> ] , <custom-ident> , <string> , <counter-style>? )
```

## \<target-text()\>

- Angle brackets are missing in syntax reference
- Last group can be omitted since [comma are implicitly omissible](https://www.w3.org/TR/css3-values/#component-types)
- A space is missing between `first-letter` and `]`

```
  target-text() = target-text( [ <string> | <url> ] [ , [ content | before | after | first-letter] ]? )
->
  <target-text()> = target-text( [ <string> | <url> ] , [ content | before | after | first-letter ]? )
```

## \<string()\>

- Angle brackets in syntax reference are missing

```
  string() = string( <custom-ident> , [ first | start | last | first-except ]? )
->
  <string()> = string( <custom-ident> , [ first | start | last | first-except ]? )
```

## \<content()\>

- Angle brackets in syntax reference are missing
- Wrong multiplier `?` is used since argument is required
- A space is missing between `[` and `text`

```
  content() = content( [text | before | after | first-letter | marker ]? )
->
  <content()> = content( [ text | before | after | first-letter | marker ] )
```

> 05/15/2017 Updated to version of 15 May 2017

**Comment by SebastianZ** [comment]
> ```
>   [ <image> ',' ]* [ normal | none | <content-list> ] [/ <string> ]?
> ->
>   [ <image> , ]* [ normal | none | <content-list> ] [/ <string> ]?
> ```

This may actually be replaced by

```
<image># [ normal | none | <content-list> ] [/ <string> ]?
```

Sebastian

**Comment by SebastianZ** [comment]
> ```
>   target-text() = target-text( [ <string> | <url> ] [ , [ content | before | after | first-letter] ]? )
> ->
>   <target-text()> = target-text( [ <string> | <url> ] [, [ content | before | after | first-letter] ]? )
> ```

Also, a space is missing between `first-letter` and `]`.

Sebastian

**Comment by lahmatiy** [comment]
> This may actually be replaced by
> ```
> <image># [ normal | none | <content-list> ] [/ <string> ]?
> ```

No, it can't: 
- `#` means `one or more times` but original syntax allows image list to be omitted, i.e. `zero or more times` should be used
- original syntax says that `<image>` list should be followed by comma, in your variant the comma after the list is prohibited

> Also, a space is missing between first-letter and ].

Agree. Thank you.

**Comment by SebastianZ** [comment]
> @SebastianZ
> 
> This may actually be replaced by
> 
>     <image># [ normal | none | <content-list> ] [/ <string> ]?
> 
> No, it can't:
> 
> - # means one or more times but original syntax allows image list to be omitted, i.e. zero or more times should be used
> - original syntax says that <image> list should be followed by comma, in your variant the comma after the list is prohibited

You're right. So, forget what I just wrote.

Sebastian

**Comment by SebastianZ** [comment]
> * No details or references about `document-url`, but looks like it's a keyword and therefore shouldn't be quoted

For reference, I've filed issue #706 asking to define `document-url`.

Sebastian

**Comment by lahmatiy** [comment]
Fixed by #1381
